server.go

```diff
@@ -83,6 +86,10 @@ func init() {
 	internal.BinaryLogger = binaryLogger
 	internal.JoinServerOptions = newJoinServerOption
 	internal.RecvBufferPool = recvBufferPool
+
+	if testing.Testing() {
+		goAwayTimeout = 0
+	}
```

Comment:
The tests that use `grpc.NewServer` and define very short timeouts should not be affected.